messages/teleporter/message_manager.go

```diff
@@ -90,14 +90,54 @@ func isAllowedRelayer(allowedRelayers []common.Address, eoa common.Address) bool
 	return false
 }
 
+// Helper to get the teleporter message from the cache or from the warp message payload
```

Comment:
```suggestion
// getTeleporterMessage returns the Warp message's corresponding Teleporter message from the cache if it exists. Otherwise parses the Warp message payload.
```

Reply:
nit: can we put this helper function below the interface implementation functions

Reply:
fixed